BlueprintUI/Sources/Blueprint View/BlueprintView.swift

```diff
@@ -75,16 +75,51 @@ public final class BlueprintView: UIView {
         fatalError("init(coder:) has not been implemented")
     }
 
-    /// Forwarded to the `measure(in:)` implementation of the root element.
+    ///
+    /// Measures the size needed to display the view within then given constraining size,
+    /// by measuring the current `element` of the `BlueprintView`.
+    ///
+    /// If you would like to not constrain the measurement in a given axis,
+    /// pass `0.0` or `.greatestFiniteMagnitude` for that axis, eg:
+    ///
+    /// ```
+    /// // Measures with a width of 100, and no height constraint.
+    /// blueprintView.sizeThatFits(CGSize(width: 100.0, height: 0.0))
+    ///
+    /// // Measures with a height of 100, and no width constraint.
+    /// blueprintView.sizeThatFits(CGSize(width: 0.0, height: 100.0))
+    ///
+    /// // Measures unconstrained in both axes.
+    /// blueprintView.sizeThatFits(.zero)
+    /// ```
+    ///
     override public func sizeThatFits(_ size: CGSize) -> CGSize {
-        guard let element = element else { return .zero }
-        let constraint: SizeConstraint
-        if size == .zero {
-            constraint = SizeConstraint(width: .unconstrained, height: .unconstrained)
-        } else {
-            constraint = SizeConstraint(size)
+        guard let element = element else {
+            return .zero
         }
-        return element.content.measure(in: constraint)
+        
+        func measurementConstraint(with size : CGSize) -> SizeConstraint {
+            
+            let unconstrainedValues : [CGFloat]
+            
+            /// Note: `.infinity` is included because `.greatestFiniteMagnitude` falls back to `.infinity` on 32 bit devices.
```

Comment:
Can someone check me on this? Without this, the tests fail on iOS 10 because we're comparing to infinity. I assume it's a 32 bit float precision thing...

Reply:
Weird, I'm going to run the tests myself but in a different project, this is `true` on 10.3.1 on iPhone 5 

```
let unconstrainedValues = [0.0, .greatestFiniteMagnitude]
print(unconstrainedValues.contains(.greatestFiniteMagnitude))
```

Reply:
Try removing the `.infinity` from here and running the tests on an iPhone 5 – they'll fail, which was why I added this. Agree, it's strange, doesn't seem quite right...

Reply:
Aha, it's because some of the tests have `CGFloat.greatestFiniteMagnitude` and some have `.greatestFiniteMagnitude` which I assume is inferring `Double` for some reason. Perhaps both 32 and 64 bit should check for `infinity`? Or should it be `CGFloat(Double.greatestFiniteMagnitude)`? This is getting weird :P

Reply:
OHHHH!! Damn, that makes a lot of sense. Guess I'll update them to reference CGFloat explicitly, since that's what we actually want.

Reply:
I think inferring `Double` happens if you use `x.0` for the other arg, since that's a double and not a CGFloat without casting explicitly. So it's probably pretty common for a size to be init'd with Double.greatestFiniteMagnitude.

Reply:
Fixed and removed the iOS version bifurcation here.

Reply:
Re initialization of CGSize: I need to test this, but CGSize does store things as CGFloat, so, maybe not a common issue?

```
    public var x: CGFloat

    public var y: CGFloat
```

(From generated Swift interface for CGSize)

Reply:
Updated to include `infinity` as well